core/roboconf-core/src/main/java/net/roboconf/core/utils/ProgramUtils.java

```diff
@@ -94,13 +96,20 @@ public static ExecutionResult executeCommandWithResult(
 
 		// Execute
 		Process process = pb.start();
+		
+		// Store process in ThreadLocal, so it can be cancelled later (eg. if blocked)
+		logger.fine("Storing process [" + applicationName + "] [" + scopedInstancePath + "]");
+		ProcessStore.setProcess(applicationName, scopedInstancePath, process);
+
 		new Thread( new OutputRunnable( process, true, errorOutput, logger )).start();
 		new Thread( new OutputRunnable( process, false, normalOutput, logger )).start();
 
 		exitValue = process.waitFor();
 		if( exitValue != 0 )
 			logger.warning( "Command execution returned a non-zero code. Code:" + exitValue );
 
+		ProcessStore.clearProcess(applicationName, scopedInstancePath);
```

Comment:
Since the method can throw exceptions, the **clearProcess** method should be invoked in a **finally** block.